src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/LinkedEditingRange/LinkedEditingRangeEndpoint.cs

```diff
@@ -25,40 +26,39 @@ internal class LinkedEditingRangeEndpoint : ILinkedEditingRangeEndpoint
         // Internal for testing only.
         internal static readonly string WordPattern = @"!?[^ <>!\/\?\[\]=""\\@" + Environment.NewLine + "]+";
 
-        private readonly DocumentContextFactory _documentContextFactory;
         private readonly ILogger _logger;
 
-        public LinkedEditingRangeEndpoint(
-            DocumentContextFactory documentContextFactory,
-            ILoggerFactory loggerFactory)
+        public LinkedEditingRangeEndpoint(ILoggerFactory loggerFactory)
         {
-            if (documentContextFactory is null)
-            {
-                throw new ArgumentNullException(nameof(documentContextFactory));
-            }
-
             if (loggerFactory is null)
             {
                 throw new ArgumentNullException(nameof(loggerFactory));
             }
 
-            _documentContextFactory = documentContextFactory;
             _logger = loggerFactory.CreateLogger<LinkedEditingRangeEndpoint>();
         }
 
-        public RegistrationExtensionResult? GetRegistration(VSInternalClientCapabilities clientCapabilities)
+        public bool MutatesSolutionState => false;
+
+        public RegistrationExtensionResult GetRegistration(VSInternalClientCapabilities clientCapabilities)
         {
             const string ServerCapability = "linkedEditingRangeProvider";
-            var option = new LinkedEditingRangeOptions { };
+            var option = new SumType<bool, LinkedEditingRangeOptions>(new LinkedEditingRangeOptions { });
 
             return new RegistrationExtensionResult(ServerCapability, option);
         }
 
-        public async Task<LinkedEditingRanges?> Handle(
-            LinkedEditingRangeParamsBridge request,
+        public TextDocumentIdentifier GetTextDocumentIdentifier(LinkedEditingRangeParams request)
+        {
+            return request.TextDocument;
+        }
+
+        public async Task<LinkedEditingRanges?> HandleRequestAsync(
+            LinkedEditingRangeParams request,
+            RazorRequestContext requestContext,
             CancellationToken cancellationToken)
         {
-            var documentContext = await _documentContextFactory.TryCreateAsync(request.TextDocument.Uri, cancellationToken).ConfigureAwait(false);
+            var documentContext = requestContext.DocumentContext;
             if (documentContext is null || cancellationToken.IsCancellationRequested)
             {
                 _logger.LogWarning("Unable to resolve document for {Uri} or cancellation was requested.", request.TextDocument.Uri);
```

Comment:
Actually, question - should we be using the logger instance within RequestContext instead? (This goes for the other PRs as well)

Reply:
So in this instance I ended up needing to have an `ILogger` on the class anyway because otherwise we'd end up passing along the `ILogger` from the RequestContext as a parameter (or worse, the whole RequestContext). I didn't want to re-pipe all that among these already massive changes so in places where sub-functions were logging I continued using the existing `ILogger`